File: leet-code/121. Best Time to Buy and Sell Stock/BestTimeToBuyAndSellStock121_2.java
public class BestTimeToBuyAndSellStock121_2 {
    public int maxProfit(int[] prices) {
        int profit = 0;
        int minPrice = prices[0];
        for (int price : prices) {
            minPrice = Math.min(minPrice, price);
            profit = Math.max(profit, price - minPrice);
        }
        return profit;
    }
}

File: leet-code/121. Best Time to Buy and Sell Stock/BestTimeToBuyAndSellStock121_2.kt
package com.example

class BestTimeToBuyAndSellStock121_2 {
    fun maxProfit(prices: IntArray): Int {
        var minPrice = prices[0]
        var maxProfit = 0
        for (i in prices.indices) {
            minPrice = minPrice.coerceAtMost(prices[i])
            maxProfit = maxProfit.coerceAtLeast(prices[i] - minPrice)
        }
        return maxProfit
    }
}
